Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -362,60 +420,81 @@ void FieldProbe::ComputeDiags (int step)
 
                 // this check is here because for m_field_probe_integrate == True, we always compute
                 // but we only write when we truly are in an output interval step
-                if (m_intervals.contains(step+1)) {
-                    for (int ip = 0; ip < np; ip++) {
+                if (m_intervals.contains(step+1))
+                {
+                    for (auto ip=numparticles - np; ip < numparticles; ip++) 
+                    {
+                        amrex::ParticleReal xp, yp, zp;
+                        getPosition(ip, xp, yp, zp);
                         // Fill output array
-                        m_data[ip * noutputs + FieldProbePIdx::Ex] = part_Ex[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Ey] = part_Ey[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Ez] = part_Ez[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Bx] = part_Bx[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::By] = part_By[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Bz] = part_Bz[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::S] = part_S[ip];
+                        m_data[ip * noutputs + 0] = ip;
```

Comment:
`std::vector::at` will do a checked access, informing you if you get out of bounds:
```suggestion
                        m_data.at(ip * noutputs + 0) = ip;
```
https://en.cppreference.com/w/cpp/container/vector/at

Reply:
You need to add the particle offset `numparticles` here